graphql/resolvers.go

```diff
@@ -3708,24 +3708,22 @@ func (r *versionResolver) TaskStatuses(ctx context.Context, v *restModel.APIVers
 }
 
 func (r *versionResolver) BaseTaskStatuses(ctx context.Context, v *restModel.APIVersion) ([]string, error) {
-	baseVersion, err := model.VersionFindOne(model.BaseVersionByProjectIdAndRevision(*v.Project, *v.Revision).WithFields(model.VersionIdentifierKey))
+	var baseVersion *model.Version
+	var err error
+
+	if !evergreen.IsPatchRequester(utility.FromStringPtr(v.Requester)) { // Get previous commit if mainline commit.
```

Comment:
Small note that this won't work for Periodic builds, since they aren't technically patches but do use the patch revision order number for whatever reason (https://github.com/evergreen-ci/evergreen/blob/698add52308336caf4e8b56f6e0e9c3e118be296/repotracker/repotracker.go#L842). I think these should actually show the base commit.